pkg/cmd/server/start/start_master.go

```diff
@@ -513,20 +515,21 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 	oc.RunSecurityAllocationController()
 
 	if kc != nil {
+		// ensure pending namespace deletions are not blocked by later controller startup failures
+		kc.RunNamespaceController()
+		// called by admission control
+		kc.RunResourceQuotaManager()
+
 		_, rcClient, err := oc.GetServiceAccountClients(oc.ReplicationControllerServiceAccount)
```

Comment:
In the openshift-infra deletion case, if there is no token, no token can be created until the namespace deletion completes, and there is actually nothing this controller can do to work successfully until the deletion completes, and another server start recreates it. I think a more fundamental question is what controllers should do if they die/fail... restart (maybe forever), die and leave other controllers running (which is what some may do today), or take down the controller process. @smarterclayton, opinions on that?